app/models/audio_file.rb

```diff
@@ -48,4 +35,24 @@ class AudioFile < BaseModel
   def fixerable_final?
     [VALID, COMPLETE, TRANSFORMING, TRANSFORM_FAILED, TRANSFORMED].include? status
   end
+
+  def compliant_with_template?
+    status_msg.nil?
+  end
+
+  def validate_on_template
```

Comment:
For some reason, this method is still bugging me.  If I just created a new audio_file ... the status is going to be UPLOADED, while we're waiting for the worker.  Will this change it to INVALID?  Will this nil out a status_msg that the lambda may have set?

@kookster you have any thoughts on this?

I _think_ the current possible state transitions (saving with a new file, background working, then validating) should look like this:

1. UPLOADED -> COMPLETE (lambda worked, template validated)
2. UPLOADED -> NOTFOUND (lambda couldn't download file)
3. UPLOADED -> FAILED (lambda couldn't process file)
4. UPLOADED -> INVALID (lambda processed file, but decided it was invalid for some reason)
5. UPLOADED -> INVALID (lambda worked, but template invalidated)

I almost think we need to nuke case (4), so we're not reusing the INVALID to mean 2 different things.

Not entirely sure how to address the overwriting-statuses/messages problem.  Anyone else have an idea?